code/modules/scripting/Scanner/Scanner.dm

```diff
@@ -100,24 +100,23 @@
 	code	 	- The source code to tokenize.
 	options - An <nS_Options> object used to configure the scanner.
 */
-/datum/n_Scanner/nS_Scanner/New(var/code, var/datum/n_scriptOptions/nS_Options/options)
+/datum/n_Scanner/nS_Scanner/New(var/list/c, var/datum/n_scriptOptions/nS_Options/options)
 	. = ..()
 	ignore += ascii2text(13) //Carriage return
 	delim += ignore + options.symbols + end_stmt + string_delim
 	src.options = options
-	LoadCode(code)
+	LoadCode(c)
 
 /datum/n_Scanner/nS_Scanner/Scan() //Creates a list of tokens from source code
 	var/list/tokens = new
-	for(, src.codepos <= length(code), src.codepos++)
-
-		var/char = copytext(code, codepos, codepos + 1)
-		var/nextchar = copytext(code, codepos + 1, codepos + 2)
+	for(, src.codepos <= code.len, src.codepos++)
+		var/char = code[codepos]
+		var/nextchar = TCOMMS_SAFE_INDEX(code, codepos + 1)
 		if(char == "\n")
 			line++
 			linepos = codepos
 
-		if(ignore.Find(char))
+		if(char in ignore)

```

Comment:
lord jesus what are you doing.
I'm fairly certain ```in``` is just as bad, if not worse than ```.Find()```

What I meant was:
```
var/list/ignore = list("\t" = TRUE, "\n" = TRUE...)
...
if(ignore[char]) continue
```

Associative lookup is very fast.